_posts/2016-09-29-Rust-1.12.md

```diff
@@ -0,0 +1,381 @@
+---
+layout: post
+title: "Announcing Rust 1.12.0"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.12.0. Rust is
+a systems programming language with the slogan "fast, reliable, productive:
+pick three."
+
+As always, you can [install Rust 1.12.0][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.12.0][notes] on GitHub.
+1361 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1120-2016-09-29
+
+### What's in 1.12.0 stable
+
+The release of 1.12 might be one of the most significant Rust releases since
+1.0. We have a lot to cover, but if you don't have time for that, here's a
+summary:
+
+The largest user-facing change in 1.12 stable is the new error message format
+emitted by `rustc`. We've [previously talked] about this format and this is the
+first stable release where they are broadly available. These error messages are
+a result of the effort of many hours of [volunteer effort] to design, test, and
+update every one of `rustc`s errors to the new format. We're excited to see
+what you think of them:
+
+![A new borrow-check error](/images/2016-09-Rust-1.12/borrowck-error.png)
+
+The largest internal change in this release is moving to a new compiler backend
+based on the new Rust [MIR]. While this feature does not result in anything
+user-visible, it paves the way for a number of future compiler optimizations,
+and for some codebases it already results in improvements to compile times and
+reductions in code size.
+
+[previously talked]: https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html
+[volunteer effort]: https://github.com/rust-lang/rust/issues/35233
+[MIR]: https://blog.rust-lang.org/2016/04/19/MIR.html
+
+#### Overhauled error messages
+
+With 1.12 we're introducing a new error format which helps to surface a lot of
+the internal knowledge about why an error is occurring to you, the developer. It
+does this by focusing on your code and the points of interest in your code that
```

Comment:
"It does this by putting your code front and center, highlighting the parts relevant to the error with annotations describing what went wrong"